harness/harness.rb

```diff
@@ -59,7 +59,16 @@ def run_benchmark(_num_itrs_hint, &block)
 
     yjit_stats&.each do |key, old_value|
       new_value = RubyVM::YJIT.runtime_stats(key)
-      diff = (new_value - old_value).to_s.reverse.scan(/\d{1,3}/).join(",").reverse
+
+      # Insert comma separators but only in the whole number portion.
+      diff = (new_value - old_value).to_s.split(".").tap do |a|
+        # Preserve any leading minus sign that may be on the beginning.
+        a[0] = a[0].reverse.scan(/\d{1,3}-?/).join(",").reverse
+        # Add a space when positive so that if there is ever a negative
+        # the first digit will line up.
+        a[0].prepend(" ") unless a[0].start_with?("-")
+      end.join(".")
```

Comment:
Might be code to put this in a utility function

Reply:
Sure.  I hesitated because we've talked before about how additional definitions in the global env can have an affect on the benchmark, but it would allow us to reuse it in the other location